analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -523,7 +523,7 @@ public void UnknownExtensionMethods()
     public void WellKnownExtensionMethods()
     {
         var list = new List<int>();
-        list.All(x => true);                                // FN
+        list.All(x => true);                                // Noncompliant
```

Comment:
It's not clear to me why, for C#, the state appears to reset after the first invocation, which prevents subsequent invocations from raising issues. 
For VB, all invocations are correctly identified
```cs
List.All(Function(X) True)      ' Noncompliant
List.Any()                      ' Noncompliant
Enumerable.Reverse(List)        ' Noncompliant
```

Reply:
Because https://github.com/SonarSource/sonar-dotnet/blob/a314cfcdfc19bf7f6e327ac8b2e78243cdcb8223/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs#L167-L171

Reply:
You'll need to fix that. Also for extensions that are well-known, and we don't raise on them.